circuit-foundation/src/commonMain/kotlin/com/slack/circuit/foundation/internal/NoOpSharedTransitionScope.kt

```diff
@@ -0,0 +1,104 @@
+package com.slack.circuit.foundation.internal
+
+import androidx.compose.animation.AnimatedVisibilityScope
+import androidx.compose.animation.BoundsTransform
+import androidx.compose.animation.EnterTransition
+import androidx.compose.animation.ExitTransition
+import androidx.compose.animation.ExperimentalSharedTransitionApi
+import androidx.compose.animation.SharedTransitionScope
+import androidx.compose.animation.SharedTransitionScope.OverlayClip
+import androidx.compose.animation.SharedTransitionScope.SharedContentState
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.graphics.addOutline
+import androidx.compose.ui.layout.LayoutCoordinates
+import androidx.compose.ui.layout.Placeable
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.LayoutDirection
+import com.slack.circuit.foundation.SharedElementTransitionLayout
+import com.slack.circuit.foundation.SharedElementTransitionScope
+import com.slack.circuit.runtime.InternalCircuitApi
+
+/**
+ * Implementation of [SharedTransitionScope] that does nothing. Used as a fallback for
+ * [SharedElementTransitionScope] when a [SharedElementTransitionLayout] hasn't been setup.
+ */
+@InternalCircuitApi
+@OptIn(ExperimentalSharedTransitionApi::class)
+public data object NoOpSharedTransitionScope : SharedTransitionScope {
```

Comment:
Curious on the need for this, vs just not adding the modifiers / layout when not required. Hard to really understand all the details of Circuit but maybe you've explored that option already?

Reply:
Was initially experimenting with this from the view `Screen` reuse regardless of the context. Thinking on it some more its not really needed, a `Screen` really shouldn't be getting reused as such.